Code/Editor/EditorFramework/Document/Implementation/GameObjectDocument.cpp

```diff
@@ -301,11 +301,22 @@ void ezGameObjectDocument::DetermineNodeName(const ezDocumentObject* pObject, co
     {
       // search for string properties that also have an asset browser property -> they reference an asset, so this is most likely the most
       // relevant property
-      if (pProperty->GetCategory() == ezPropertyCategory::Member &&
-          (pProperty->GetSpecificType() == ezGetStaticRTTI<const char*>() || pProperty->GetSpecificType() == ezGetStaticRTTI<ezString>()) &&
-          pProperty->GetAttributeByType<ezAssetBrowserAttribute>() != nullptr)
+      if (
+        (pProperty->GetSpecificType() == ezGetStaticRTTI<const char*>() || pProperty->GetSpecificType() == ezGetStaticRTTI<ezString>()) && pProperty->GetAttributeByType<ezAssetBrowserAttribute>() != nullptr)
       {
-        ezStringBuilder sValue = pChild->GetTypeAccessor().GetValue(pProperty->GetPropertyName()).ConvertTo<ezString>();
+        ezStringBuilder sValue;
+        if (pProperty->GetCategory() == ezPropertyCategory::Member)
+        {
+          sValue = pChild->GetTypeAccessor().GetValue(pProperty->GetPropertyName()).ConvertTo<ezString>();
+        }
+        else if (pProperty->GetCategory() == ezPropertyCategory::Array)
+        {
+          const ezInt32 iCount = pChild->GetTypeAccessor().GetCount(pProperty->GetPropertyName());
+          if (iCount > 0)
+          {
+            sValue = pChild->GetTypeAccessor().GetValue(pProperty->GetPropertyName(), 0).ConvertTo<ezString>();
+          }
+        }
```

Comment:
what is this change about ?

Reply:
"Decals and other components that have arrays of assets will not have an auto-generated name in the scene graph referencing the first entry in the asset array."
Basically as the decal uses an array of assets it was not picked up so all decals were called "Decal". With this change the name now looks like this:
![image](https://user-images.githubusercontent.com/11784812/211376778-5bd1b5ec-f760-4a83-8976-37b884825bf7.png)